### PR TITLE
Handle blank diagnostic codes for request issues from the front end

### LINKED ISSUE #10948: AttorneyJudgeCheckoutError: Diagnostic code is not included in the list
Validation failed: Benefit type is not included in the list

https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/4971/

### PR DESCRIPTION
**Pull request by hschallhorn**
Resolves #10948

### Description
Diagnostic code is not required on the [front end](https://github.com/department-of-veterans-affairs/caseflow/pull/9825) or on the [back end](https://github.com/department-of-veterans-affairs/caseflow/blob/43562ed9e75b14c6f949802521da9df6a4214c2b/app/models/decision_issue.rb#L10-L11) for decision issues. However, when diagnostic code is not included, the back end receives an empty string, not a nil, causing us to throw a `Caseflow::Error::AttorneyJudgeCheckoutError: Validation failed: Diagnostic code is not included in the list` to [sentry](https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/9479/events/2bcdd75a2f7648cab9998ada98a0002d/). This PR fixes that error but returning nil for blank diagnostic codes.

### Acceptance Criteria
- [ ] Empty diagnostic codes passed to case review controller are correctly handled by making them nil

### Testing Plan
1. Was unable to replicate this is testing as my dev environment always passes nil when completing an attorney case review with a blank diagnostic code 🤷

### CODE REVIEW

File: app/models/concerns/issue_updater.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@ def create_decision_issues!
         disposition: issue_attrs[:disposition],
         description: issue_attrs[:description],
         benefit_type: issue_attrs[:benefit_type],
-        diagnostic_code: issue_attrs[:diagnostic_code],
+        diagnostic_code: issue_attrs[:diagnostic_code].presence,
```
Comment (hschallhorn) [inline app/models/concerns/issue_updater.rb:47]: Actualy change! Shorthand for `issue_attrs[:diagnostic_code].blank? ? nil : issue_attrs[:diagnostic_code]` or `issue_attrs[:diagnostic_code].present? ? issue_attrs[:diagnostic_code] : nil`

File: spec/controllers/case_reviews_controller_spec.rb [hunk 2]
```diff
@@ -19,6 +19,44 @@
...
+            expect(request_issue2.decision_issues.first.remand_reasons.first.post_aoj).to eq true
+
+            expect(task.reload.status).to eq "completed"
+            expect(task.closed_at).to_not eq nil
+            expect(task.parent.reload.status).to eq "assigned"
+            expect(task.parent.type).to eq JudgeDecisionReviewTask.name
+
+            expect(bva_dispatch_task_count_before).to eq(BvaDispatchTask.count)
+          end
+        end
```
Comment (hschallhorn) [inline spec/controllers/case_reviews_controller_spec.rb:58]: Pulled out of ["when all parameters are present" test](https://github.com/department-of-veterans-affairs/caseflow/pull/14030/files#diff-fc0fc85f102646ebc88ee93a7570ec5fL154)

File: spec/controllers/case_reviews_controller_spec.rb [hunk 3]
```diff
@@ -27,38 +65,40 @@
...
+                         "description": "great moments",
+                         "benefit_type": "vha",
+                         "diagnostic_code": "5002",
+                         "request_issue_ids": [request_issue2.id],
+                         "remand_reasons": [{ "code": "va_records", "post_aoj": true }] }]
+          }
+        end
+
+        subject { post :complete, params: { task_id: task.id, tasks: params }, as: :json }
+
```
Comment (hschallhorn) [inline spec/controllers/case_reviews_controller_spec.rb:95]: Pulled out repeating params code to get those lines deleted numbers up 😤

File: spec/controllers/case_reviews_controller_spec.rb [hunk 4]
```diff
@@ -27,38 +65,40 @@
...
+                       { "disposition": "remanded",
+                         "description": "great moments",
+                         "benefit_type": "vha",
+                         "diagnostic_code": "5002",
+                         "request_issue_ids": [request_issue2.id],
+                         "remand_reasons": [{ "code": "va_records", "post_aoj": true }] }]
+          }
+        end
+
+        subject { post :complete, params: { task_id: task.id, tasks: params }, as: :json }
```
Comment (hschallhorn) [inline spec/controllers/case_reviews_controller_spec.rb:94]: Pulled out repeating `params` and `post` to get those lines deleted numbers up 😤

File: app/models/concerns/issue_updater.rb [hunk 5]
```diff
@@ -44,7 +44,7 @@ def create_decision_issues!
         disposition: issue_attrs[:disposition],
         description: issue_attrs[:description],
         benefit_type: issue_attrs[:benefit_type],
```
Comment (yoomlam) [inline app/models/concerns/issue_updater.rb:46]: Wondering if `benefit_type` should be validated.

**Pull request merged**